Fix tagging of release candidate images for helm

mimowo opened the pull request:
#### What type of PR is this?

/kind bug

#### What this PR does / why we need it:

#### Which issue(s) this PR fixes:
<!--
*Automatically closes linked issue when PR is merged.
Usage: `Fixes #<issue number>`, or `Fixes (paste link of issue)`.
_If PR is about `failing-tests or flakes`, please post the related issues/tests in a comment and do not use `Fixes`_*
-->
Fixes #

#### Special notes for your reviewer:

#### Does this PR introduce a user-facing change?
<!--
If no, just write "NONE" in the release-note block below.
If yes, a release note is required:
Enter your extended release note in the block below. If the PR requires additional action from users switching to the new release, include the string "action required".
-->
```release-note
NONE
```

mimowo commented:
/assign @tenzen-y

mimowo commented:
/retest

tenzen-y commented on hack/push-chart.sh:
```diff
@@ -32,7 +32,7 @@ HELM=${HELM:-./bin/helm}
 
 readonly k8s_registry="registry.k8s.io/kueue"
-readonly semver_regex='^v([0-9]+)(\.[0-9]+){1,2}$'
+readonly semver_regex='^v([0-9]+)(\.[0-9]+)(-rc)?(\.[0-9]+)?$'
```
I want to align with the semver regexp since the kueue dedicated semver regexp means maintaining it by ourselves. Otherwise, we can delegate the maintenance to the semver specification.

mimowo replied:
Updated tentatively use the official regex, but I had to prefix it with "v".

tenzen-y replied:
> Also, I'm not sure the intention of the var is to match semvar. For example, we require "v" prefix, but this does not match the official semvar prefix anyway.

Yes, right. The semver does not include `v`. However, I think using the v + semver regexp makes this script maintainable as opposed to defining the original regexp.

mimowo replied:
/hold
we don't need named groups, let me use the shorter version without those

tenzen-y replied:
+1

mimowo replied:
Actually, it does not seem the syntax is supported in bash, I get error `failed to compile regex: Invalid preceding regular expression`

mimowo replied:
@tenzen-y I reverted the regex to the one custom. I understand your intention, but it seems the regex is not compatible fully with bash. We can open follow up issue.

tenzen-y replied:
What about `
if echo "${EXTRA_TAG}" | grep -Pq "${semver_regex}"; then`

mimowo replied:
> What about ` if echo "${EXTRA_TAG}" | grep -Pq "${semver_regex}"; then`

Yes, this passed my testing. I'm ok with that. PTAL

tenzen-y reviewed (commented):
Otherwise lgtm
Thank you

tenzen-y commented on hack/push-chart.sh:
```diff
@@ -32,7 +32,8 @@ HELM=${HELM:-./bin/helm}
 
 readonly k8s_registry="registry.k8s.io/kueue"
-readonly semver_regex='^v([0-9]+)(\.[0-9]+){1,2}$'
+# Regex from: https://regex101.com/r/Ly7O1x/3/, but with "v" prefix
```
```suggestion
# Regex from: https://semver.org/#is-there-a-suggested-regular-expression-regex-to-check-a-semver-string, but with "v" prefix
```

Shouldn't we use the semantic versioning official documentation?

tenzen-y reviewed (commented):
Awesome! Thank you
/lgtm
/approve

mimowo commented:
/test pull-kueue-test-e2e-tas-main
infra failure 

```
Switched to context "kind-kind".
error: unable to read URL "https://raw.githubusercontent.com/kubeflow/mpi-operator/v0.6.0/deploy/v2beta1/mpi-operator.yaml", server reported 429 Too Many Requests, status code=429
Switched to context "kind-kind".
Exporting logs for cluster "kind" to:
/logs/artifacts/run-test-tas-e2e-1.32.3
No resources found in kueue-system namespace.
```

mimowo commented:
/test pull-kueue-test-e2e-main-1-30
